checkpoint.go

```diff
@@ -36,7 +36,8 @@ type CheckParams struct {
 	Version string
 
 	// Generic product flags
-	Flags map[string]string
+	Flags      map[string]string
+	ExtraFlags func() []Flag
```

Comment:
Now that we have a more "tolerant" data structure (i.e. sequence of pairs as opposed to map) with `ExtraFlags func() []Flag`, is there a point keeping `Flags      map[string]string` at all?